noxfile.py

```diff
@@ -0,0 +1,279 @@
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_build_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def docker_qa_build_version(session: "nox.session", version: str) -> None:
+    set_environment(session, version, False)
+    session.run(
+        "bash",
+        "-c",
+        " ".join(
+            [
+                "docker",
+                "build",
+                "--tag",
+                docker_container_id(version),
+                "--build-arg",
+                "USERID={}".format(os.geteuid()),
+                "--build-arg",
+                "CC=${CC}",
+                "--build-arg",
+                "CXX=${CXX}",
+                "--file",
+                "admin/Dockerfile.qa",
+                ".",
+            ]
+        ),
+        external=True,
+    )
+
+
+@nox.session
+def docker_qa_run(session: "nox.session") -> None:
+    session_id = "docker_qa_run_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def docker_qa_run_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_run_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def docker_qa_run_version(session: "nox.session", version: str) -> None:
+    set_environment(session, version, False)
+    session.run(
+        "bash",
+        "-c",
+        " ".join(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-e",
+                "TESTOPTS",
+                "-e",
+                "CC",
+                "-v",
+                "{}:/gcovr".format(os.getcwd()),
+                docker_container_id(version),
+            ]
+        ),
+        external=True,
+    )
+
+
+@nox.session
+def docker_qa(session: "nox.session") -> None:
+    session_id = "docker_qa_build_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+    session_id = "docker_qa_run_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def docker_qa_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_build_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+        session_id = "docker_qa_run_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
```

Comment:
Can replace loop contents with `session.notify("docker_qa_version({version})")`

Reply:
You're right.